sdk/test/AzureTest/Scrubbing.swift

```diff
@@ -0,0 +1,164 @@
+// --------------------------------------------------------------------------
+//
+// Copyright (c) Microsoft Corporation. All rights reserved.
+//
+// The MIT License (MIT)
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the ""Software""), to
+// deal in the Software without restriction, including without limitation the
+// rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
+// sell copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED *AS IS*, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+// FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
+// IN THE SOFTWARE.
+//
+// --------------------------------------------------------------------------
+
+import Foundation
+
+public class Scrubbing {
+    
+    //regex for subscription id scrubbing
+    static let subscriptionIDRegex = try! NSRegularExpression(pattern: "(/(subscriptions))/[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}", options: [.caseInsensitive])
+    static let graphSubscriptionIDRegex = try! NSRegularExpression(pattern: "(https://(graph.windows.net))/[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}", options: [.caseInsensitive])
+    
+    static func scrubSubscriptionIDs(request: URLRequest) -> URLRequest {
+        
+        var cleanRequest = request
+        
+        let dirtyHeaders = request.allHTTPHeaderFields ?? [:]
+        var cleanHeaders : [String:String] = [:]
+        for key in dirtyHeaders.keys {
+            cleanHeaders[key] = regexSubscriptionReplace(dirtyHeaders[key])
+        }
+        if cleanHeaders.isEmpty {
+            cleanRequest.allHTTPHeaderFields = nil
+        } else {
+            cleanRequest.allHTTPHeaderFields = cleanHeaders
+        }
+        
+        cleanRequest.url = URL(string: regexSubscriptionReplace(request.url?.absoluteString))
+        cleanRequest.mainDocumentURL = URL(string: regexSubscriptionReplace(request.mainDocumentURL?.absoluteString))
+        cleanRequest.httpBody = regexSubscriptionReplace(String(data: request.httpBody, encoding: .utf8))?.data(using: .utf8)
+    
+        return cleanRequest
+    }
+    
+    static func regexSubscriptionReplace(_ string: String?)->String?{
```

Comment:
This should probably not be static, should be more descriptive, and avoid using `_`:
```suggestion
    func scrubSubscriptionId(from string: String?) -> String? {
```